app/models/name/parse.rb

```diff
@@ -323,7 +323,7 @@ def parse_genus_or_up(str, deprecated = false, rank = "Genus")
 
   def parse_below_genus(str, deprecated, rank, pattern)
     results = nil
-    if (match = pattern.match(str))
+    if (match = pattern.to_s.match?(str))
```

Comment:
```suggestion
    if (match = pattern.match(str))
```
We need the `match` method here because the subsequent lines expect the `match` local variable to be a MatchData object, rather than true/false.
See https://ruby-doc.org/3.1.2/Regexp.html#method-i-match